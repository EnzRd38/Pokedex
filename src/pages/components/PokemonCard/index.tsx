import { useContext } from "react"
import Favorites from "../../contexts/favorites"

const PokemonCard = (props: { pokemon: any }) => {
	const {favoritePokemons, updateFavoritePokemons} = useContext(Favorites)
	const { pokemon } = props
	const favoriteClick = () => {
		updateFavoritePokemons(pokemon.name)
	}
	const heart = favoritePokemons.includes(pokemon.name) ? "‚ù§Ô∏è" : "üñ§"

	return (
	<div className="pokemon-card">
		<div className="pokemon-image-container">
			<img alt={pokemon.name} src={pokemon.sprites.front_default} className="pokemon-image" />
		</div>
		<div className="card-body">
			<div className="card-top">
				<h3> {pokemon.name} </h3>
				<h4>#{pokemon.id}</h4>
			</div>
			<div className="card-bottom">
				<div className="pokemon-types">
					{pokemon.types.map((type, index) =>{
						return (
							<div key={index} className="pokemon-type-name">
								{type.type.name}
							</div>
						)
					})}
				</div>
				<button className="pokemon-favorite" onClick={favoriteClick}>
					{heart}
				</button>
			</div>
		</div>
	</div>
	)
}

export default PokemonCard
