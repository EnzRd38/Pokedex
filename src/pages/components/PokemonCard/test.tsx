import { render, screen } from '@testing-library/react'
import PokemonCard from '.'

describe('<PokemonCard />', () => {
	it('should render the heading', () => {
		const { container } = render(<PokemonCard />)

		expect(
			screen.getByRole('heading', { name: /react avan√ßado/i })
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
