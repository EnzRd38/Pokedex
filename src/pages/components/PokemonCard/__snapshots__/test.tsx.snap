// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonCard /> should render the heading 1`] = `
<main>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
